reporters/dogstatsd.go

```diff
@@ -9,19 +9,27 @@ package reporters
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
+	"github.com/DataDog/datadog-go/statsd"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 	"github.com/topfreegames/extensions/dogstatsd"
-	constants "github.com/topfreegames/maestro/reporters/constants"
+	"github.com/topfreegames/maestro/reporters/constants"
 	handlers "github.com/topfreegames/maestro/reporters/dogstatsd"
 )
 
 // DogStatsD reports metrics to a dogstatsd.Client
 type DogStatsD struct {
-	client dogstatsd.Client
-	logger *logrus.Logger
-	region string
+	client       dogstatsd.Client
+	logger       *logrus.Logger
+	statsdClient *statsd.Client
+	mutex        sync.Mutex
```

Comment:
Can you use a RWMutex? That way we allow multiple reads but only one write.

Reply:
Changed to RWMutex